runtime/src/main/java/org/corfudb/protocols/service/CorfuProtocolMessage.java

```diff
@@ -145,6 +165,14 @@ public static ResponseMsg getResponseMsg(HeaderMsg header, ResponsePayloadMsg re
      * @return        a ResponseMsg with the provided header and error
      */
     public static ResponseMsg getResponseMsg(HeaderMsg header, ServerErrorMsg error) {
+        if (header.getVersion().getCorfuSourceCodeVersion() !=
+                GitRepositoryState.getCorfuSourceCodeVersion()) {
```

Comment:
Is this the best place for such a check? The assumption here was that the response header was already built.

Reply:
Will do this in a following cleanup/refactor PR.

Reply:
+1 @zfrenette  comment. I think it should be addressed in this PR.

Reply:
Done, thanks!